src/lib/cache/subCache.cpp

```diff
@@ -690,57 +683,33 @@ void subCacheItemInsert(CachedSubscription* cSubP)
 */
 void subCacheItemInsert
 (
-  const char*                   tenant,
-  const char*                   servicePath,
-  const ngsiv2::HttpInfo&       httpInfo,
-  const EntityIdVector&         entityIdVector,
-  const AttributeList&          attributeList,
-  const NotifyConditionVector&  notifyConditionVector,
-  const char*                   subscriptionId,
-  int64_t                       expirationTime,
-  int64_t                       throttling,
-  RenderFormat                  renderFormat,
-  bool                          notificationDone,
-  int64_t                       lastNotificationTime,
-  StringFilter*                 stringFilterP,
-  StringFilter*                 mdStringFilterP,
-  const std::string&            status,
-  const std::string&            q,
-  const std::string&            geometry,
-  const std::string&            coords,
-  const std::string&            georel,
-  bool                          blacklist
+  const char*                      tenant,
+  const char*                      servicePath,
+  const ngsiv2::HttpInfo&          httpInfo,
+  const EntityIdVector&            entityIdVector,
+  const AttributeList&             attributeList,
+  const std::vector<std::string>&  notifyConditionV,
+  const char*                      subscriptionId,
+  int64_t                          expirationTime,
+  int64_t                          throttling,
+  RenderFormat                     renderFormat,
+  bool                             notificationDone,
+  int64_t                          lastNotificationTime,
+  StringFilter*                    stringFilterP,
+  StringFilter*                    mdStringFilterP,
+  const std::string&               status,
+  const std::string&               q,
+  const std::string&               geometry,
+  const std::string&               coords,
+  const std::string&               georel,
+  bool                             blacklist
 )
 {
   //
   // Add the subscription to the subscription cache.
-  // But only if any of the entities in entityIdVector is pattern based -
-  // AND it is a subscription of type ONCHANGE
-  //
-
+  // But only if any of the entities in entityIdVector is pattern based - is this true?
```

Comment:
In fact, I guess that the subscrpition refresh logic is taking all the subscriptions or many .test will fail...